src/data/real/basic.lean

```diff
@@ -603,6 +603,55 @@ by rw [mul_comm, sqrt_mul' _ hx, mul_comm]
 @[simp] theorem sqrt_div (hx : 0 ≤ x) (y : ℝ) : sqrt (x / y) = sqrt x / sqrt y :=
 by rw [division_def, sqrt_mul hx, sqrt_inv]; refl
 
+/--
+This is used in proving the lemma that `sqrt` is a continuous function.
+The lemma, called `real.continuous_sqrt`, is located at the file `analysis.specific_limits`
+-/
+lemma abs_sqrt_sub_sqrt_le_sqrt_abs (x y : ℝ) : abs (sqrt x - sqrt y) ≤ sqrt (abs (x - y)) :=
+nonneg_le_nonneg_of_squares_le (sqrt_nonneg _) $
+begin
+  rw abs_mul_abs_self,
+  wlog h : y ≤ x using [x y, y x],
+  -- The case goal, i.e. the permutation covers all possible cases:
+  { exact le_total y x },
+  -- The main goal, i.e., prove the goal with `y ≤ x`
+  { have eq₁ : sqrt (abs (x - y)) * sqrt (abs (x - y)) = x - y,
+      rw mul_self_sqrt (abs_nonneg _), apply abs_of_nonneg, linarith,
+    rw eq₁,
+    have eq₂ : (sqrt x - sqrt y) * (sqrt x - sqrt y) =
+      sqrt x * sqrt x - (2 * sqrt x * sqrt y - sqrt y * sqrt y), ring,
+    rw eq₂,
+    cases le_total 0 x with hx hx,
```

Comment:
```suggestion
    by_cases hx : 0 ≤ x,
```

Reply:
That's not quite the same @robertylewis: `le_total` has a nice symmetry to it while with `by_cases` you'll have to get rid of a negation

Reply:
I like `cases le_total 0 x with hx hx` better. It gets me `hx : 0 ≤ x` and `hx : x ≤ 0`.

`by_cases hx : 0 ≤ x` produces `¬0 ≤ x`, which is not easy to use. 
(For example, `theorem sqrt_eq_zero_of_nonpos (h : x ≤ 0) : sqrt x = 0` needs `x ≤ 0`, which is two steps away from `¬0 ≤ x`)